File: Anul 2/Metode Avansate de Programare/Seminarii/sem11-taskrunner/taskrunner/taskrunner/runner/StrategyTaskRunner.cs
using taskrunner.container;
using taskrunner.factory;

namespace taskrunner.runner;

public class StrategyTaskRunner : ITaskRunner
{
    private IContainer container;

    public StrategyTaskRunner(Strategy strategy)
    {
        container = TaskContainerFactory.GetInstance().CreateContainer(strategy);
    }

    public void ExecuteOneTask()
    {
        if (!container.IsEmpty())
        {
            Task task = container.Remove();
            if (task != null)
            {
                task.Execute();
            }
        }
    }

    public void ExecuteAll()
    {
        while (!container.IsEmpty())
        {
            ExecuteOneTask();
        }
    }

    public void AddTask(Task t)
    {
        container.Add(t);
    }

    public bool HasTask()
    {
        return !container.IsEmpty();
    }
}

File: Anul 2/Metode Avansate de Programare/Laboratoare/lab-1-Mothyque/src/runner/StrategyTaskRunner.java
package runner;

import domain.Container;
import domain.Strategy;
import domain.Task;
import factory.TaskContainerFactory;

public class StrategyTaskRunner implements TaskRunner
{
    private Container container;

    public StrategyTaskRunner(Strategy strategy)
    {
        this.container = TaskContainerFactory.getInstance().createContainer(strategy);
    }

    @Override
    public void executeOneTask()
    {
        if (!container.isEmpty())
        {
            Task task = container.remove();
            if (task != null)
            {
                task.execute();
            }
        }
    }

    @Override
    public void executeAll()
    {
        while (!container.isEmpty())
        {
            executeOneTask();
        }
    }

    @Override
    public void addTask(Task task)
    {
        container.add(task);
    }

    @Override
    public boolean hasTask()
    {
        return !container.isEmpty();
    }
}
